File: app/src/main/java/com/luxlingo/app/model/SeedModels.kt
package com.luxlingo.app.model

import com.google.gson.annotations.SerializedName

data class InitialSeedData(
    val vocabulary: List<SeedVocab>,
    val senses: List<SeedSense>,
    val sentences: List<SeedSentence>,
    val curriculum: List<SeedLesson>
)

data class SeedVocab(
    @SerializedName("surface_id") val surfaceId: String,
    @SerializedName("lemma_id") val lemmaId: String,
    @SerializedName("word_lu") val wordLu: String,
    @SerializedName("audio_ref") val audioRef: String
)

data class SeedSense(
    @SerializedName("sense_id") val senseId: String,
    @SerializedName("surface_id") val surfaceId: String,
    @SerializedName("primary_en") val primaryEn: String,
    val pos: String,
    @SerializedName("is_golden_key") val isGoldenKey: Boolean = false,
    @SerializedName("is_picturable") val isPicturable: Boolean = false
)

data class SeedSentence(
    @SerializedName("sentence_id") val sentenceId: String,
    @SerializedName("text_lu") val textLu: String,
    @SerializedName("text_en") val textEn: String,
    @SerializedName("sense_ids") val senseIds: List<String>,
    @SerializedName("cloze_index") val clozeIndex: Int
)

data class SeedLesson(
    @SerializedName("lesson_id") val lessonId: String,
    @SerializedName("title_en") val titleEn: String,
    @SerializedName("core_senses") val coreSenses: List<String>,
    @SerializedName("secondary_senses") val secondarySenses: List<String> = emptyList()
)

File: ios/LuxLingo/LuxLingo/Models/SeedModels.swift
import Foundation

// MARK: - Seed Data (parsed from initial_seed.json)

struct InitialSeedData: Codable {
    let version: Double
    let vocabulary: [SeedVocab]
    let senses: [SeedSense]
    let sentences: [SeedSentence]
    let curriculum: [SeedLesson]
    let articleExercises: [ArticleExerciseSeed]?
    let bonusLessons: [BonusSeedLesson]?

    enum CodingKeys: String, CodingKey {
        case version, vocabulary, senses, sentences, curriculum
        case articleExercises = "article_exercises"
        case bonusLessons = "bonus_lessons"
    }
}

struct SeedVocab: Codable {
    let surfaceId: String
    let lemmaId: String
    let wordLu: String
    let audioRef: String
    let lodAudioUrl: String?

    enum CodingKeys: String, CodingKey {
        case surfaceId = "surface_id"
        case lemmaId = "lemma_id"
        case wordLu = "word_lu"
        case audioRef = "audio_ref"
        case lodAudioUrl = "lod_audio_url"
    }
}

struct SeedParadigm: Codable {
    let present: [String]
}

struct SeedSense: Codable {
    let senseId: String
    let surfaceId: String
    let primaryEn: String
    let pos: String
    let isGoldenKey: Bool?
    let isPicturable: Bool?
    let paradigm: SeedParadigm?

    enum CodingKeys: String, CodingKey {
        case senseId = "sense_id"
        case surfaceId = "surface_id"
        case primaryEn = "primary_en"
        case pos
        case isGoldenKey = "is_golden_key"
        case isPicturable = "is_picturable"
        case paradigm
    }
}

struct SeedSentence: Codable {
    let sentenceId: String
    let textLu: String
    let textEn: String
    let senseIds: [String]
    let clozeIndex: Int
    let difficulty: String
    let nRuleWordIndex: Int?
    let nRuleForm: String?
    let exactForm: Bool?
    let clozeConfidence: String?

    enum CodingKeys: String, CodingKey {
        case sentenceId = "sentence_id"
        case textLu = "text_lu"
        case textEn = "text_en"
        case senseIds = "sense_ids"
        case clozeIndex = "cloze_index"
        case difficulty
        case nRuleWordIndex = "n_rule_word_index"
        case nRuleForm = "n_rule_form"
        case exactForm = "exact_form"
        case clozeConfidence = "cloze_confidence"
    }
}

struct SeedLesson: Codable {
    let lessonId: String
    let titleEn: String
    let coreSenses: [String]
    let secondarySenses: [String]?

    enum CodingKeys: String, CodingKey {
        case lessonId = "lesson_id"
        case titleEn = "title_en"
        case coreSenses = "core_senses"
        case secondarySenses = "secondary_senses"
    }
}

struct ArticleExerciseSeed: Codable {
    let id: String
    let senseId: String
    let textLu: String
    let textEn: String
    let articleIndex: Int
    let correct: String
    let options: [String]
    let ruleHint: String
    let difficulty: String

    enum CodingKeys: String, CodingKey {
        case id, senseId = "sense_id", textLu = "text_lu", textEn = "text_en"
        case articleIndex = "article_index", correct, options
        case ruleHint = "rule_hint", difficulty
    }
}

struct BonusSeedLesson: Codable {
    let lessonId: String
    let titleEn: String
    let situationTag: String
    let unitIndex: Int
    let sceneImage: String
    let coreSenses: [String]

    enum CodingKeys: String, CodingKey {
        case lessonId = "lesson_id", titleEn = "title_en"
        case situationTag = "situation_tag", unitIndex = "unit_index"
        case sceneImage = "scene_image", coreSenses = "core_senses"
    }
}
